frontend/catalyst/jax_tracer.py

```diff
@@ -405,25 +412,123 @@ def trace_observables(obs, qubit_states, p, num_wires, qreg):
     elif isinstance(obs, qml.operation.Tensor):
         nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs.obs]
         jax_obs = jprim.tensorobs(*nested_obs)
     elif isinstance(obs, qml.Hamiltonian):
         nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs.ops]
         jax_obs = trace_hamiltonian(op_args, *nested_obs)
-    else:
+    elif paulis := obs._pauli_rep:  # pylint: disable=protected-access
+        # Use the pauli sentence representation of the observable, if applicable
+        jax_obs = pauli_sentence_to_rt_obs(paulis, qubit_states, qreg)
+    elif isinstance(obs, qml.ops.op_math.Sum):
+        nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs]
+        op_args = jax.numpy.ones(len(obs))
+        jax_obs = trace_hamiltonian(op_args, *nested_obs)
+    elif isinstance(obs, qml.ops.op_math.Prod):
+        nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs]
+        jax_obs = jprim.tensorobs(*nested_obs)
+    elif isinstance(obs, qml.ops.op_math.SProd):
+        terms = obs.terms()
+        coeffs = jax.numpy.array(terms[0])
+        nested_obs = trace_observables(terms[1][0], qubit_states, p, num_wires, qreg)[0]
+        jax_obs = jprim.hamiltonian(coeffs, nested_obs)
+    else:  # pragma: no cover
         raise RuntimeError(f"unknown observable in measurement process: {obs}")
     return jax_obs, qubits
 
 
+def named_obs(obs, op_args, qubit_states, qreg):
+    """Trace a named observable.
+
+    Args:
+        obs: a named observable
+        op_args: arguments of the observable
+        qubit_states: the statically known qubit state at this program point
+        qreg: the quantum register with the state at this program point
+
+    Returns:
+        jax_obs: a NamedObs JAX primitive used for tracing
+        qubits: a list of qubits used by the observable
+    """
+    assert isinstance(obs, KNOWN_NAMED_OBS)
+    _, wires = op_args
+    qubits = get_qubits_from_wires(wires, qubit_states, qreg)
+    jax_obs = jprim.namedobs(type(obs).__name__, qubits[0])
+    return jax_obs, qubits
+
+
+def hermitian_obs(obs, op_args, qubit_states, qreg):
+    """Trace a Hermitian observable.
+
+    Args:
+        obs: a Hermitian observable
+        op_args: arguments of the observable
+        qubit_states: the statically known qubit state at this program point
+        qreg: the quantum register with the state at this program point
+
+    Returns:
+        jax_obs: a Hermitian JAX primitive used for tracing
+        qubits: a list of qubits used by the observable
+    """
+    assert isinstance(obs, qml.Hermitian)
+    matrix, wires = op_args
+    qubits = get_qubits_from_wires(wires, qubit_states, qreg)
+    jax_obs = jprim.hermitian(matrix, *qubits)
+    return jax_obs, qubits
+
+
+def pauli_sentence_to_rt_obs(paulis, qubit_states, qreg):
```

Comment:
Minor but I think it's clearer :)
```suggestion
def pauli_sentence_to_hamiltonian(paulis, qubit_states, qreg):
```